src/cpp/benders/benders_by_batch/include/BatchCollection.h

```diff
@@ -8,22 +9,48 @@
 struct Batch {
   std::vector<std::string> sub_problem_names;
   unsigned id;
+  friend class boost::serialization::access;
+  template <class Archive>
+  void serialize(Archive &ar, const unsigned int version) {
+    ar &sub_problem_names;
+    ar &id;
+  }
 };
 class BatchCollection {
  private:
+  std::vector<std::string> sub_problem_names_;
```

Comment:
Do we really need to keep the `sub_problem_names_` as attributes here, as they are stored again in the vector of `Batch` ?

Reply:
this attribute is used to build batches see BuildBatches method, otherwise batches must be built inside the ctor...